ckan/controllers/feed.py

```diff
@@ -145,7 +145,7 @@ def _create_atom_id(resource_path, authority_name=None, date_string=None):
         # This is best we can do, and if the site_url is not set, then
         # this still results in an invalid feed.
         site_url = config.get('ckan.site_url', '')
-        return '/'.join([site_url, resource_path])
+        return urlparse.urljoin(site_url, resource_path)
```

Comment:
still not quite right. We need to use url_for here so that the url is correctly built including site_url and root_path settings. This fix will only work for sites without a root_path setting